pkg/auth/auth.go

```diff
@@ -56,7 +56,10 @@ func AuthBackend(shard conn.DBInstance, berule *config.BackendRule, msg pgproto3
 		if rule == nil {
 			return fmt.Errorf("auth rule not set for %s-%s-%s", shard.ShardName(), berule.DB, berule.Usr)
 		}
-
+		username := berule.Usr
+		if rule != nil && rule.Usr != "" {
```

Comment:
```suggestion
		if rule.Usr != "" {
```
`rule != nil` is redundant